src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -167,11 +170,16 @@ async def load_lockfile(
             ),
         )
         requirement_estimate = _pex_lockfile_requirement_count(lock_bytes)
+        constraints_strings = None
     else:
         header_delimiter = "#"
+        lock_string = lock_bytes.decode()
         # Note: this is a very naive heuristic. It will overcount because entries often
         # have >1 line due to `--hash`.
-        requirement_estimate = len(lock_bytes.decode().splitlines())
+        requirement_estimate = len(lock_string.splitlines())
+        constraints_strings = FrozenOrderedSet(
+            str(req) for req in parse_requirements_file(lock_string, rel_path=lockfile_path)
+        )
```

Comment:
I don't think that this will handle hash correctly?

Reply:
Indeed: added another commit to improve parsing.